pkgs/tools/misc/toybox/default.nix

```diff
@@ -0,0 +1,67 @@
+{
+  stdenv, lib, fetchFromGitHub, buildPackages,
+  enableStatic ? false,
+  enableMininal ? false,
+  extraConfig ? ""
+}:
+stdenv.mkDerivation rec {
+  name = "toybox-${version}";
+  version = "0.8.0";
+
+  src = fetchFromGitHub {
+    owner = "landley";
+    repo = "toybox";
+    rev = "b11e1206b4193dd46b4c40c333175fceadd409df";
+    sha256 = "0pghxf2ihx79rmqqgg61850m6kcnck51nn1ng5x20vd0wfh6gdl6";
+  };
+
+  depsBuildBuild = [ buildPackages.stdenv.cc ];
+  buildInputs = lib.optionals enableStatic [ stdenv.cc.libc stdenv.cc.libc.static ];
+
+  postPatch = "patchShebangs scripts";
+
+  configurePhase = ''
+    make ${if enableMininal then "allnoconfig" else if stdenv.isFreeBSD then "freebsd_defconfig" else if stdenv.isDarwin then "macos_defconfig" else "defconfig"}
+
+    function parseconfig {
+      while read LINE; do
+        NAME="$(echo "$LINE" | sed -e 's/^\([^ =]*\).*$/\1/')"
+        OPTION="$(echo "$LINE" | sed -e 's/[^ =]*[ =]\([^ =]*\)$/\1/')"
+
+        if ! [[ "$NAME" =~ ^CONFIG_ ]]; then continue; fi
+
+        echo "parseconfig: removing $NAME"
+        sed -i /$NAME'\(=\| \)'/d .config
+
+        if [ "$OPTION" = 'n' ]; then
+          echo "parseconfig: unsetting $NAME"
+          echo "# $NAME is not set" >> .config
+        else
+          echo "parseconfig: setting $NAME=$OPTION"
+          echo "$NAME=$OPTION" >> .config
+        fi
+      done
+    }
+
+    cat << EOF | parseconfig
+    ${extraConfig}
+    EOF
+
+    make oldconfig
+  '';
+
+  makeFlags = [ "PREFIX=$(out)" ] ++ stdenv.lib.optional enableStatic "LDFLAGS=--static";
+
+  # tests currently (as of 0.8.0) get stuck in an infinite loop
+  #doCheck = true;
+  checkTarget = "tests";
+
+  meta = with stdenv.lib; {
+    description = "Lightweight implementation of some Unix command line utilities";
+    homepage = "http://landley.net/toybox/";
+    license = licenses.bsd0;
+    platforms = with platforms; linux ++ darwin ++ freebsd;
+    maintainers = [ maintainers.hhm ];
+    priority = 10;
```

Comment:
Is there a reason this is set? Pretty sure this shouldn't be here.

Reply:
`nixos-rebuild` uses `toybox`'s utils if it has the same priority as coreutils. Before version 0.8.0, `toybox`'s `mktemp` did not have the `-t` flag, which was causing `nixos-rebuild` to fail. `nixos-rebuild` may work with 0.8.0 (did not have chance to test yet), but figured I should set a lower priority just in case there are still some errors with core Nix(OS) tools (ideally I suppose perhaps `nixos-rebuild` should explicitly depend on GNU `coreutils`?), although the user can do this manually too.

Reply:
Oh interesting, yeah it would probably be best if `nixos-rebuild` depended on `coreutils` explicitly, that would be a nice PR.

I get the argument that we don't want to have the user have problems because they use a different set of such utilities. But on the other hand they kind of ask for it if they install toybox to begin with, they wouldn't install it to use the standard `coreutils` commands.

I think for now it's fine to keep this priority low for toybox because this is the safer default, we can always remove this later. If the user really wants to use the toybox versions, they can override the priority.

Reply:
Ok sounds good. Thanks!